include/Reader/readerir.h

```diff
@@ -1484,16 +1501,21 @@ class GenIR : public ReaderBase {
   /// \returns    LLVM type that models the built-in object type.
   llvm::Type *getBuiltInObjectType();
 
-  /// Get the LLVM type for an array of references.
+  /// Get the LLVM type for an array of given element type.
   ///
   /// Used when we know that some type must be an array but our local
   /// type information thinks otherwise.
   ///
-  /// \returns    LLVM type that models array of references.
-  llvm::PointerType *getArrayOfReferenceType();
+  /// \param ElementTy                 The desired element type for the array, 
+  ///                                  or if null, use array of references.
+  /// \returns                         LLVM type that models the desired array
+  llvm::PointerType *getArrayOfElementType(llvm::Type *ElementTy = nullptr);
 
-  /// Create the LLVM type for an array of references.
-  void createArrayOfReferenceType();
+  /// \brief Creates the type of the array with specified element type.
+  /// 
+  /// \param ElementTy The llvm element type that is desired for the array type.
+  /// \returns The LLVM representation of the MSIL array type.
+  llvm::PointerType *createArrayOfElementType(llvm::Type *ElementTy = nullptr);
```

Comment:
My preference is for ElementTy not to be allowed to be nullptr. It should be Object if an array of references is asked for.